src/NuGet.Core/NuGet.Packaging/Signing/Verification/SignatureTrustAndValidityVerificationProvider.cs

```diff
@@ -95,11 +96,14 @@ private PackageVerificationResult VerifyValidityAndTrust(Signature signature, Si
             {
                 using (var authorSignatureNativeCms = NativeCms.Decode(signature.SignedCms.Encode(), detached: false))
                 {
-                    var signatureHash = NativeCms.GetSignatureValueHash(signature.SignatureContent.HashAlgorithm, authorSignatureNativeCms);
-
-                    if (!IsTimestampValid(timestamp, signatureHash, treatIssuesAsErrors, failIfUnknownRevocation, issues) && treatIssuesAsErrors)
+                    if (!allowIgnoreTimestamp)
```

Comment:
`AllowIgnoreTimestamp` is supposed to be used as "I allow that if a timestamp is not valid just ignore it instead of failing".... By doing this you are purposefully ignoring the timestamp, which is not the intent of this flag. Should we rename it to something more descriptive?